emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -35,207 +17,114 @@ def setUpClass(self):
-        Executes a function with timing, stores the timing or error data, and verifies the storage.
-
-        Parameters:
-        - name (str): The name of the function or code block.
-        - func (Callable[[], bool]): The function to execute and time.
-        - store_success (Callable): Function to store timing data on success.
-        - store_error (Callable): Function to store error data on failure.
-        """
-        logging.info(f"Starting timing for '{name}'.")
-
-        # Mapping of storage functions and corresponding keys
-        storage_mapping = {
-            'success': (store_success, "stats/dashboard_time"),
-            'error': (store_error, "stats/dashboard_error")
-        }
-
-        try:
-            with ec_timer.Timer() as timer:
-                result = func()  # Execute the function
-
-            elapsed_ms = timer.elapsed * 1000  # Convert to milliseconds
-
-            # Store the execution time using the appropriate storage function
-            store_func, key = storage_mapping['success']
-            store_func(code_fragment_name=name, timer=timer)
-            logging.info(f"'{name}' executed successfully in {elapsed_ms:.2f} ms.")
-
-            # Verify the timing entry
-            self.verify_entry(key, name, elapsed_ms)
-
-        except Exception as e:
-            # Capture elapsed time up to the exception
-            elapsed_seconds = timer.elapsed if 'timer' in locals() else 0
-            elapsed_ms = elapsed_seconds * 1000  # Convert to milliseconds
-
-            # Store the error timing using the appropriate storage function
-            store_func, key = storage_mapping['error']
-            store_func(code_fragment_name=name, timer=timer)
-            logging.error(f"'{name}' failed after {elapsed_ms:.2f} ms with error: {e}")
-
-            # Verify the error entry
-            self.verify_entry(key, name, elapsed_ms)
-
-            # Re-raise the exception to let the test fail
-            raise
+        edb.get_timeseries_db().delete_many(
+            {"metadata.key": {"$in": keys_to_clean}}
+        )
+        logging.debug(f"After test, cleared DB entries for {keys_to_clean}")
 
-    def verify_entry(self, key: str, name: str, expected_elapsed_ms: float):
+    def verify_stats_entries(self, expected_entries: list[tuple[str, str, float]]):
         """
-        Verifies that an entry for the given name exists under the specified key and matches the expected elapsed time.
+        Verifies that each of these expected entries, in the form of (key, name, elapsed_ms),
+        are stored in the database.
 
-        Parameters:
-        - key (str): The database key to query ("stats/dashboard_time" or "stats/dashboard_error").
-        - name (str): The name of the function or code block.
-        - expected_elapsed_ms (float): The expected elapsed time in milliseconds.
+        :param expected_entries: A list of tuples which have (key, name, expected_elapsed_ms).
         """
```

Comment:
`verify_entry` was good because it reduced the repeated lines of code that you had in a previous version.
However, this approach prevents you from checking for the existence of multiple entries, which is something Shankari asked for.

You need to be able to check for multiple expected entries, so I changed this to accept a list and renamed it to `verify_stats_entries`
After it queries the timeseries for relevant keys, it iterates and compares stored vs. expected values at each index